packages/core-components/src/layout/Sidebar/Bar.tsx

```diff
@@ -78,10 +92,12 @@ enum State {
 type Props = {
   openDelayMs?: number;
   closeDelayMs?: number;
+  disableExpandOnHover?: boolean;
```

Comment:
isPinned=false still allows expanding on hover
I think disableExpandOnHover provides slightly different functionality - it prevents the sidebar from expanding completely unless it is pinned or an expand button is added to control it via a click instead of mouseover.